internal/common/rules/ruleutilities.go

```diff
@@ -1,9 +1,9 @@
 package rules
 
 // PickUpRulesByVariable returns a list of rule_id's which are affected by certain variables.
-func PickUpRulesByVariable(variableName VariableFieldName, ruleStore map[string]RuleMatrix) ([]string, bool) {
+func PickUpRulesByVariable(variableName VariableFieldName, ruleStore map[string]RuleMatrix, variableMap map[VariableFieldName]VariableValuePair) ([]string, bool) {
 	var Rules []string
-	if _, ok := VariableMap[variableName]; ok {
+	if _, ok := variableMap[variableName]; ok {
 		for k, v := range ruleStore {
 			_, found := searchForVariableInArray(variableName, v.RequiredVariables)
```

Comment:
Did we not refactor some searches to return -1 when not found? 
@E-Mac27

Reply:
anyway, not within the scope of this diff :)

Reply:
I believe so. We technically dont need the bool but meh.